bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/magento/GraphqlDataServiceImpl.java

```diff
@@ -262,28 +264,36 @@ private CategoryTree getCategoriesImpl(Integer categoryId, String storeView) {
         return category;
     }
 
+    private String toCategoryCacheKey(Integer categoryId, Integer currentPage, Integer pageSize) {
+        return StringUtils.join(categoryId, currentPage, pageSize);
```

Comment:
You should add a separator here to avoid ambiguities. For example to separate category 11 on page 1 and category 1 on page 11.

Reply:
Good point, I fixed it.